src/pyqasm/visitor.py

```diff
@@ -2228,3 +2286,286 @@ def finalize(self, unrolled_stmts):
+
+    def _validate_calibration_statement_dispatch(self, stmt, gate_name: str, definition: pulse_ast.CalibrationDefinition) -> None:
+        """Route calibration statements to EXISTING validators."""
+        stmt_type_name = type(stmt).__name__
+        
+        if stmt_type_name == 'ClassicalDeclaration':
+            # Only validate openpulse-specific types, then use existing validator
+            self._validate_openpulse_declaration_type(stmt, gate_name, definition)
+            # Try to use existing validator if compatible
+            try:
+                self._visit_classical_declaration(stmt)
+            except (AttributeError, TypeError):
+                # If pulse_ast isn't compatible, do basic validation
+                self._basic_pulse_declaration_validation(stmt, gate_name, definition)
+                
+        elif stmt_type_name == 'ClassicalAssignment':
+            # Use existing assignment validator
+            try:
+                self._visit_classical_assignment(stmt)
+            except Exception as err:
+                raise_qasm3_error(
+                    f"Invalid assignment in calibration definition for '{gate_name}': {err}",
+                    error_node=definition,
+                    span=definition.span,
+                    raised_from=err,
+                )
+                
+        elif stmt_type_name in ['BranchingStatement', 'IfStatement']:
+            # Use existing branching validator
+            try:
+                self._visit_branching_statement(stmt)
+            except Exception as err:
+                raise_qasm3_error(
+                    f"Invalid branching statement in calibration definition for '{gate_name}': {err}",
+                    error_node=definition,
+                    span=definition.span,
+                    raised_from=err,
+                )
+                
+        elif stmt_type_name in ['ForInLoop', 'ForLoop']:
+            # Use existing loop validator
+            try:
+                self._visit_forin_loop(stmt)
+            except Exception as err:
+                raise_qasm3_error(
+                    f"Invalid loop in calibration definition for '{gate_name}': {err}",
+                    error_node=definition,
+                    span=definition.span,
+                    raised_from=err,
+                )
+                
+        elif stmt_type_name == 'ExpressionStatement':
+            # Validate openpulse function calls
+            self._validate_openpulse_function_call(stmt, gate_name, definition)
+            
+        else:
+            raise_qasm3_error(
+                f"Unsupported statement type '{stmt_type_name}' in calibration definition for '{gate_name}'",
+                error_node=definition,
+                span=definition.span,
+            )
+
+    def _get_calibration_arg_size(self, arg_type) -> int:
+        """Get size for calibration argument type."""
+        type_name = type(arg_type).__name__
+        return {
```

Comment:
Move the dict to `maps/pulse.py` and directly use the `.get` method on it. Will ensure that we're not redefining it again and again during fn call.